src/functions/org/apache/jmeter/functions/StringToFile.java

```diff
@@ -76,6 +76,7 @@ public StringToFile() {
     private boolean writeToFile() throws IOException {
         String fileName = ((CompoundVariable) values[0]).execute().trim();
         String content = ((CompoundVariable) values[1]).execute();
+        content = setLineSeparatorByDifferentSystems(content);    
```

Comment:
I prefer the addition of a new parameter to let user decide if he wants a newline or not (defaulting to false). Doing such replacements might lead to bugs in different OSes. 
Also add a Junit test.
Thank you

Reply:
About the bug, it's because of the "reserved character" in Java; for example if the user entres a "\r\n", it will become "\\\\r\\\\n" ,so the lineseperator can't run normally